unistd/dir.c

```diff
@@ -247,8 +255,11 @@ static int _resolve_abspath(char *path, char *result, int resolve_last_symlink,
 		}
 
 		/* prepend resolved symlink to remaining unresolved path */
-		p -= 1;
-		*p = '/';
+		if (*p != '\0') {
```

Comment:
is this a related change? is there any test for a case this is needed?

Reply:
Yes. Previously if there was a path like `/path/to/symlink` and `symlink` had contents `/realpath`, after fetching symlink next iteration would look up `/realpath/`. It didn't matter because trailing slashes were always ignored during lookup. This change is necessary to prevent adding the slash when not necessary (when symlink is the final part of the path).

Adding the slash is necessary when looking up something like `/path/to/symlink/file_in_dir`, because we need to look up `/realpath/file_in_dir`.

No additional test cases are necessary, it is already covered by looking up a symlink to a file and trying to traverse through symlink to directory.